cli/command/node/list.go

```diff
@@ -45,21 +45,29 @@ func newListCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runList(dockerCli *command.DockerCli, opts listOptions) error {
 	client := dockerCli.Client()
+	out := dockerCli.Out()
 	ctx := context.Background()
 
+	var info types.Info
+
 	nodes, err := client.NodeList(
 		ctx,
 		types.NodeListOptions{Filter: opts.filter.Value()})
 	if err != nil {
 		return err
 	}
 
-	info, err := client.Info(ctx)
+	if len(nodes) == 0 {
+		// if there is no node, no need to call /api endpoint and print directly.
+		goto SKIP
+	}
+
+	info, err = client.Info(ctx)
```

Comment:
Thanks for your review. @aaronlehmann 
I do not think goto is good, either. Now I updated this part, PTAL.